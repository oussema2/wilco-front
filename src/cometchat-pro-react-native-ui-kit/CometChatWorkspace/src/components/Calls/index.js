export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
