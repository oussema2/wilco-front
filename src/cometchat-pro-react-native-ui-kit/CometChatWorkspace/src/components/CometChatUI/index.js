/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
