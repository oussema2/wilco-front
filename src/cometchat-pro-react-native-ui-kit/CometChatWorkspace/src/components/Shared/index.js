export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
