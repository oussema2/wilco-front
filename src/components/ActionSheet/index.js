export { default as ActionSheet } from './ActionSheet';
