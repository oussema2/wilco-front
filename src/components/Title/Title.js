import React from 'react';
import { Text, View, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import { styles } from './Title.styles';
import noop from '../../helpers/noop';

const Title = ( {
	testID, text, textTransform, style, rightText, rightAction, required
} ) => (
	<View style={{ flexDirection: 'row', flex: 1 }}>
		<Text
			testID={testID}
			style={{ ...styles( { textTransform } ).text, ...style }}
			numberOfLines={1}
		>
			{text}
		</Text>
		{required && <Text testID="required-testID" style={{ ...styles( { textTransform } ).required }}> *</Text> }
		{ rightText && rightAction
					&& (
						<Text
							testID="right-text-testID"
							onPress={rightAction}
							style={{
								...styles( { textTransform } ).text,
								...style,
								...styles( { textTransform } ).rightText
							}}
							numberOfLines={1}
						>
							{rightText}
						</Text>
					)}
	</View>
);

Title.propTypes = {
	testID: PropTypes.string,
	text: PropTypes.string.isRequired,
	textTransform: PropTypes.oneOf( [ 'capitalize', 'none' ] ),
	style: ViewPropTypes.style,
	rightText: PropTypes.string,
	rightAction: PropTypes.func,
	required: PropTypes.bool
};

Title.defaultProps = {
	testID: 'title-component',
	textTransform: 'capitalize',
	style: {},
	rightAction: noop,
	rightText: null,
	required: false
};

export default Title;
