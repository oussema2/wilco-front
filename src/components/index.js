export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
