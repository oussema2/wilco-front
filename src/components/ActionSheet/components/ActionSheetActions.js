import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
