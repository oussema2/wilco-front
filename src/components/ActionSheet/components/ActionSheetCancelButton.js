import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
