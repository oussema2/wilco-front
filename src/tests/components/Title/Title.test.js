import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import { Title } from '../../../components/Title';
import noop from '../../../helpers/noop';

afterEach( cleanup );

describe( 'Title', () => {
	const testID = 'testing-title-component';
	const text = 'A title';

	describe( 'with the default props', () => {
		it( 'renders the Title component correctly', () => {
			const component = render( <Title testID={testID} text={text} /> );

			expect( component.queryByTestId( testID ) ).toHaveTextContent( text );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with a required prop', () => {
		it( 'renders the Title component correctly', () => {
			const component = render( <Title testID={testID} text={text} required /> );
			expect( component.queryByTestId( 'required-testID' ) ).toBeDefined();
			expect( component.queryByTestId( 'required-testID' ) ).toHaveTextContent( '*' );
		} );
	} );

	describe( 'with a rightText prop', () => {
		it( 'renders right text correctly', () => {
			const exampleText = 'example-testID';
			const component = render( <Title
				testID={testID}
				text={text}
				rightText={exampleText}
				rightAction={noop}
			/> );
			expect( component.queryByTestId( 'right-text-testID' ) ).toBeDefined();
			expect( component.queryByTestId( 'right-text-testID' ) ).toHaveTextContent( exampleText );
		} );
	} );

	describe( 'without a rightText prop', () => {
		it( 'does not render the right text', () => {
			const component = render( <Title
				testID={testID}
				text={text}
				rightAction={noop}
			/> );
			expect( component.queryByTestId( 'right-text-testID' ) ).toBeNull();
		} );
	} );
} );
