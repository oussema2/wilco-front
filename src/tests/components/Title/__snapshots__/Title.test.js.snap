// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title with the default props renders the Title component correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "NotoSansJP-Black",
        "fontSize": 16,
        "fontWeight": undefined,
        "letterSpacing": -0.3,
        "lineHeight": 22,
        "minHeight": 22,
        "textTransform": "capitalize",
      }
    }
    testID="testing-title-component"
  >
    A title
  </Text>
</View>
`;
